### ISSUE TITLE
A question about q loss

### ISSUE THREAD

**Issue by wang1946may7** [baturaysaglam/AC-Off-POC#1]
Hi, thanks for providing us code. Actually, I ran the code, it seems that for contininous action, 
the critic loss , 
 critic_loss = torch.sum(js_weights * (F.mse_loss(current_q1, target_q, reduction='none') + F.mse_loss(current_q2, target_q, reduction='none'))) / torch.sum(js_weights)

it seems that js_weight is a 1d tensor with all same value?  as you did js_div.item() * torch.ones_like(reward)

you use a 1d  js_weight to multiply q loss, ant then divided by  torch.sum(js_weights). I am not sure it is meaningful to do this. 
Really hope your kind response. Best!

**Comment by baturaysaglam** [comment]
did you run it for a deterministic policy or stochastic one?

**Comment by wang1946may7** [comment]
Hi, I ran for continuous mujoco control, so i think is determistic one, as the finnal layer only output the mean of each action

**Comment by baturaysaglam** [comment]
yes, it is deterministic policies. actually you're right, you don't have to treat _js_weight_ like a tensor because the weight value is always the same for a sampled batch. I just used the same line of code that is also used in SAC (stochastic policy) there. you can also do:

```
critic_loss = js_weights * (F.mse_loss(current_q1, target_q) + F.mse_loss(current_q2, target_q))
```

**Comment by wang1946may7** [comment]
critic_loss = torch.sum(js_weights * (F.mse_loss(current_q1, target_q, reduction='none') + F.mse_loss(current_q2, target_q, reduction='none'))) / torch.sum(js_weights)
equals to
critic_loss =(F.mse_loss(current_q1, target_q) + F.mse_loss(current_q2, target_q)) I think, 
not 
critic_loss = js_weights * (F.mse_loss(current_q1, target_q) + F.mse_loss(current_q2, target_q))

your js_weights not work in critic?

**Comment by wang1946may7** [comment]
I do think 
critic_loss = js_weights * (F.mse_loss(current_q1, target_q) + F.mse_loss(current_q2, target_q))
is good, 
but it did not work properly in your code
